core/src/main/java/jenkins/security/AgentSecretAction.java

```diff
@@ -0,0 +1,78 @@
+package jenkins.security;
```

Comment:
This code is more related to working with agents even if it deals with a connection secret. `jenkins.agents` would be appropriate IMO.